cpp/src/Ice/ConnectionFactory.cpp

```diff
@@ -338,6 +338,20 @@ IceInternal::OutgoingConnectionFactory::~OutgoingConnectionFactory()
     assert(_pendingConnectCount == 0);
 }
 
+ObjectAdapterPtr
+IceInternal::OutgoingConnectionFactory::getDefaultObjectAdapter() const noexcept
+{
+    lock_guard lock(_mutex);
+    return _defaultObjectAdapter;
+}
+
+void
+IceInternal::OutgoingConnectionFactory::setDefaultObjectAdapter(ObjectAdapterPtr adapter) noexcept
+{
+    lock_guard lock(_mutex);
+    _defaultObjectAdapter = dynamic_pointer_cast<ObjectAdapterI>(adapter);
```

Comment:
Should we assert the cast succeed?

Reply:
There is no situation where this dynamic_pointer_cast can fail.

If you're concerned about users creating their own ObjectAdapter, we can make ObjectAdapter's constructor protected.